src/modules/form/components/DynamicForm.tsx

```diff
@@ -115,42 +126,43 @@ const DynamicForm = forwardRef(
       }
     }, []);
 
-    const { renderFields, getCleanedValues } = useDynamicFields({
-      definition,
-      initialValues,
-      localConstants,
-      onFieldChange,
-    });
+    const { renderFields, getCleanedValues, getValuesForSubmit } =
```

Comment:
Maybe, add a comment here explaining why `getCleanedValues` is returned separately from the hook even though it's also called by `getValuesForSubmit`? My understanding is that `getCleanedValues` alone is needed by the `MciClearance` component and that's why it's passed to the context here. It seems like `MciClearance` also calls `createHudValuesForSubmit` internally too, so maybe it would be desirable to clean this up further by just making `MciClearance` receive `getValuesForSubmit` from the context instead of `getCleanedValues`?

It was confusing to me that we refer to the result of `getCleanedValues` as `rawValues`, which is part of why I'm wondering if it's possible to further clean up -- maybe if `getCleanedValues` were only internal to the `useDynamicFields` hook it would be easier to follow. 

The comments you've added showing how these objects look are really helpful, thanks 🙏

Reply:
Thanks, good point. I renamed `getCleanedValues` to `getValues` since the values are not really "cleaned", they just have the hidden fields dropped. I don't want to mess too much with the MCI Clearance component, so I left that function in the context. I also added a comment here explaining the usage of getValues vs getValuesForSubmit.